src/rules/style/end_statements.rs

```diff
@@ -0,0 +1,220 @@
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{ASTRule, Rule, Violation};
+use tree_sitter::Node;
+
+pub struct AbbreviatedEndStatement {}
+
+impl Rule for AbbreviatedEndStatement {
+    fn new(_settings: &Settings) -> Self {
+        Self {}
+    }
+
+    fn explain(&self) -> &'static str {
+        "
+        End statements should specify what kind of construct they're ending, and the
+        name of that construct. For example, prefer this:
+
+        ```
+        module mymodule
+          ...
+        end module mymodule
+        ```
+
+        To this:
+
+        ```
+        module mymodule
+          ...
+        end
+        ```
+
+        Or this:
+
+        ```
+        module mymodule
+          ...
+        end module
+        ```
+
+        Similar rules apply for submodules, programs, functions, subroutines, and
+        derived types.
+        "
+    }
+}
+
+fn map_start_statement(kind: &str) -> &'static str {
```

Comment:
_Could_ squash all three functions into one and return a tuple?

Reply:
Done in my latest changes, and I simplified it a little so that I only need two outputs instead of three.